src/alerting/components/builder/ThresholdCondition.tsx

```diff
@@ -51,17 +51,9 @@ const ThresholdCondition: FC<Props> = ({
   onUpdateThreshold,
   onRemoveThreshold,
 }) => {
-  const [inputs, changeInputs] = useState([
-    get(threshold, 'value') || get(threshold, 'min', 0),
-    get(threshold, 'max', 100),
-  ])
-
-  const min = get(threshold, 'value') || get(threshold, 'min', inputs[0])
-  const max = get(threshold, 'max', inputs[1])
-
-  useEffect(() => {
-    changeInputs([min, max])
-  }, [min, max])
+  const inputs = []
+  inputs[0] = get(threshold, 'value') || get(threshold, 'min', 0)
+  inputs[1] = get(threshold, 'max', 100)
```

Comment:
this is an area I'd appreciate some eyes.

The original code looked pretty circular to me and as far as I can tell, this is equivalent, but avoids an infinite re-render.